src/wrapper/clap-juce-wrapper.cpp

```diff
@@ -1843,20 +1905,123 @@ class ClapJuceWrapper : public clap::helpers::Plugin<
+
+        if (scale > 50)
+        {
+            // this is almost definitely a units error
+            scale *= 0.01;
+        }
+        guiScaleFactor = static_cast<float>(scale);
+
+        if (editorWrapper != nullptr)
+        {
+            editorWrapper->setEditorScaleFactor(guiScaleFactor);
+            return true;
+        }
+
+        return true;
+    }
+
+    /*
+     * guiAdjustSize is called before guiSetSize and given the option to
+     * reset the size the host hands to the subsequent setSize. This is a
+     * relatively naive and unsatisfactory initial implementation.
+     */
+    bool guiAdjustSize(uint32_t *w, uint32_t *h) noexcept override
+    {
+        std::cout << "guiADJUST SIZE " << *w << " " << *h << std::endl;
+        if (editorWrapper == nullptr || editorWrapper->editor == nullptr)
+            return false;
+
+        if (!editorWrapper->editor->isResizable())
+            return false;
+
+        auto cst = editorWrapper->editor->getConstrainer();
+
+        if (!cst)
+            return true; // we have no constraints. Whaever is fine!
+
+        const auto minBounds = EditorWrapperComponent::convertToHostBounds(
+            {cst->getMinimumWidth(), cst->getMinimumHeight()});
+        const auto maxBounds = EditorWrapperComponent::convertToHostBounds(
+            {cst->getMaximumWidth(), cst->getMaximumHeight()});
+        auto minW = (uint32_t)minBounds.getWidth();
+        auto maxW = (uint32_t)maxBounds.getWidth();
+        auto minH = (uint32_t)minBounds.getHeight();
+        auto maxH = (uint32_t)maxBounds.getHeight();
+
+        // There is no std::clamp in c++14
+        auto width = juce::jlimit(minW, maxW, *w);
+        auto height = juce::jlimit(minH, maxH, *h);
+
+        auto aspectRatio = (float)cst->getFixedAspectRatio();
+
+        if (aspectRatio > 0.0f)
+        {
+            /*
+             * This is obviously an unsatisfactory algorithm, but we wanted to have
+             * something at least workable here.
+             *
+             * The problem with other algorithms I tried is that this function gets
+             * called by BWS for sub-single pixel motions on macOS, so it is hard to make
+             * a version which is *stable* (namely adjust(w,h); cw=w;ch=h; adjust(cw,ch);
+             * cw == w; ch == h) that deals with directions. I tried all sorts of stuff
+             * and then ran into vacation.
+             *
+             * So for now here's this approach. See the discussion in CJE PR #67
+             * and interop-tracker issue #30.
+             */
+            width = (uint32_t)std::round(aspectRatio * (float)height);
+        }
+
+        *w = width;
+        *h = height;
+        std::cout << "End of GAdj " << width << " " << height << std::endl;
+
+        return true;
+    }
+
+    bool guiSetSize(uint32_t width, uint32_t height) noexcept override
+    {
+        std::cout << "GUI SET SIZE " << width << " " << height << std::endl;
```

Comment:
ditto. maybe actually we want a `TRACE_GUI_SIZE` macro left here so we can debug later? so like

```
#define TRACE_GUI_SIZE(x)
// #define TRACE_GUI_SIZE(x) std::cout << __FILE__ <<":" << __LINE__ << " " << x << std::endl;
```